src/pages/individual/Individual.jsx

```diff
@@ -46,30 +50,56 @@ export default function Individual() {
   const history = useHistory();
   const fieldSchemas = useIndividualFieldSchemas();
 
-  function refreshIndividualData() {
-    const queryKey = getIndividualQueryKey(id);
-    queryClient.invalidateQueries(queryKey);
-  }
+  const refreshIndividualData = useCallback(
+    () => {
+      const queryKey = getIndividualQueryKey(id);
+      queryClient.invalidateQueries(queryKey);
+    },
+    [queryClient],
+  );
 
   const metadata = useMemo(
     () => {
       if (!individualData || !fieldSchemas) return null;
-      return fieldSchemas.map(schema => ({
-        ...schema,
-        value: schema.getValue(schema, individualData),
-      }));
+      return fieldSchemas.map(schema => {
+        const augmentedSchema = {
+          ...schema,
+          value: schema.getValue(schema, individualData),
+        };
+
+        if (schema.name === 'defaultName') {
+          augmentedSchema.nameGuid = deriveIndividualNameGuid(
+            individualData,
+            'defaultName',
+          );
+        } else if (schema.name === 'nickname') {
+          augmentedSchema.nameGuid = deriveIndividualNameGuid(
+            individualData,
+            'nickname',
+          );
+        }
+
+        return augmentedSchema;
+      });
     },
     [individualData, fieldSchemas],
   );
 
-  const names = individualData?.names || [];
-  const defaultNameObject = names.find(
-    n => n.context === 'defaultName',
+  const [defaultName, nickname] = useMemo(
+    () => [
+      deriveIndividualName(
+        individualData,
+        'defaultName',
+        'Unnamed individual',
+      ),
+      deriveIndividualName(
+        individualData,
+        'nickname',
+        'Unnamed individual',
+      ),
```

Comment:
if there is no nickname, let's just leave off the nickname text rather than call it "unnamed individual"

Reply:
There is already a separate ticket for this (DEX-844), so I'm leaving this as is, since it preserves the current behavior.